MdePkg/Include/Protocol/UsbAssociationIo.h

```diff
@@ -0,0 +1,201 @@
+/** @file
+  EFI Usb Interface Association protocol.
+
+  This protocol is used by USB drivers, running in the EFI boot services
+  environment to access USB devices that implement their configurations
+  using interface association: USB cameras, USB audio devices, USB modems, etc.
+
+  Copyright (c) 2024, American Megatrends International LLC. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+

```

Comment:
Which specification is this Protocol defined?  Is there a PPI as well?

Reply:
This is the new protocol introduction, prototyped in this PR.

Reply:
If it is not part of an industry standard  spec or a public spec, then it can not go into the MdePkg.

Reply:
It is a part of USB specification from 3.0 onwards.

Reply:
The UEFI Protocol is defined in the USB Specification?

Reply:
Of course not :) USB data structures - descriptors, interfaces, - are defined in the UEFI spec, they are referenced. Usb.h has definitions from USB specifications. Interface Association Descriptor definition was missing.

Reply:
Did you mean MdePkg only has definitions from UEFI spec so USB association protocol has to move to MdeModulePkg? Please confirm my understanding is correct.

Reply:
No. Protocols in MdePkg need to be defined in an industry standard or public spec. What section of the USB Specification is this protocol defined?  That needs to be in the file header for code review.  If it is not defined in an industry standard or public spec, then it is being defined in the EDK II implementation.  Those types of protocols may be defined in the MdeModulePkg.

Reply:
Protocol defined in UsbAssociationIo.h is not in any public spec. The idea was to present it to UEFI community and have it  in the future UEFI specification update. Thank you for clarification, will move this protocol to MdeModulePkg.